pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_statement_hint_builtins failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_statement_hint_builtins [failed](https://mesolite.cluster.engflow.com/invocations/default/811dbf37-3921-4652-945f-60d0a5a53796?testReportRun=1&testReportShard=31&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL2xvY2FsOmxvY2FsX3Rlc3Q=) on master @ [4442354d04be76f56cbbdb06f76894fb3dcf4fbf](https://github.com/cockroachdb/cockroach/commits/4442354d04be76f56cbbdb06f76894fb3dcf4fbf):


```
                 skip reason: index "abc_c_idx" not found
            
        but found (query options: "") :
            planning time: 10µs
            execution time: 100µs
            distribution: <hidden>
            vectorized: <hidden>
            plan type: generic, re-optimized
            statement hints: 1 (0 applied, 1 skipped)
            maximum memory usage: <hidden>
            DistSQL network usage: <hidden>
            regions: <hidden>
            isolation level: serializable
            priority: normal
            quality of service: regular
            ·
            • project
            │ columns: (c)
            │
            └── • scan
                  columns: (a, c)
                  sql nodes: <hidden>
                  kv nodes: <hidden>
                  regions: <hidden>
                  vectorized batch count: 0
                  KV time: 0µs
                  KV rows decoded: 0
                  KV pairs read: 0
                  KV bytes read: 0 B
                  KV gRPC calls: 0
                  estimated max memory allocated: 0 B
                  MVCC step count (ext/int): 0/0
                  MVCC seek count (ext/int): 0/0
                  actual row count: 0
                  estimated row count: 1 (missing stats)
                  table: abc@abc_pkey
                  spans: /10-/11
            ·
            skipped statement hints: 1
             └── REWRITE INLINE HINTS
                 donor: SELECT c FROM abc@abc_c_idx WHERE a = _
                 skip reason: index "abc_c_idx" not found
            
[17:10:54] --- done: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins with config local: 254 tests, 1 failures
    logic.go:4688: 
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins:2370: error while processing
    logic.go:4688: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins:2370: too many errors encountered, skipping the rest of the input
    panic.go:694: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_statement_hint_builtins2621415134
--- FAIL: TestExecBuild_statement_hint_builtins (7.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=31</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_statement_hint_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63354

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_statement_hint_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330250?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330250?buildTab=artifacts#/) on master @ [8e517daeefe190c460334c5a97f235775f5072ee](https://github.com/cockroachdb/cockroach/commits/8e517daeefe190c460334c5a97f235775f5072ee):


```
                 skip reason: index "abc_c_idx" not found
            
        but found (query options: "") :
            planning time: 10µs
            execution time: 100µs
            distribution: <hidden>
            vectorized: <hidden>
            plan type: generic, re-optimized
            statement hints: 1 (0 applied, 1 skipped)
            maximum memory usage: <hidden>
            DistSQL network usage: <hidden>
            regions: <hidden>
            isolation level: serializable
            priority: normal
            quality of service: regular
            ·
            • project
            │ columns: (c)
            │
            └── • scan
                  columns: (a, c)
                  sql nodes: <hidden>
                  kv nodes: <hidden>
                  regions: <hidden>
                  vectorized batch count: 0
                  KV time: 0µs
                  KV rows decoded: 0
                  KV pairs read: 0
                  KV bytes read: 0 B
                  KV gRPC calls: 0
                  estimated max memory allocated: 0 B
                  MVCC step count (ext/int): 0/0
                  MVCC seek count (ext/int): 0/0
                  actual row count: 0
                  estimated row count: 1 (missing stats)
                  table: abc@abc_pkey
                  spans: /10-/11
            ·
            skipped statement hints: 1
             └── REWRITE INLINE HINTS
                 donor: SELECT c FROM abc@abc_c_idx WHERE a = _
                 skip reason: index "abc_c_idx" not found
            
[19:22:01] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20353/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins with config local: 254 tests, 1 failures
    logic.go:4688: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20353/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins:2370: error while processing
    logic.go:4688: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20353/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins:2370: too many errors encountered, skipping the rest of the input
    panic.go:694: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/36236bfe7e0683829cc7a3dd6f3a0eaa/logTestExecBuild_statement_hint_builtins4076095282
--- FAIL: TestExecBuild_statement_hint_builtins (8.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_statement_hint_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins
+++ b/pkg/sql/opt/exec/execbuilder/testdata/statement_hint_builtins
@@ -2309,15 +2309,15 @@ SELECT crdb_internal.await_statement_hints_cache()
 
 statement ok
 SELECT information_schema.crdb_rewrite_inline_hints(
-  'SELECT c FROM abc WHERE a = _',
-  'SELECT c FROM abc@abc_c_idx WHERE a = _'
+  'SELECT a FROM abc WHERE a = _',
+  'SELECT a FROM abc@abc_c_idx WHERE a = _'
 )
 
 statement ok
 SELECT crdb_internal.await_statement_hints_cache()
 
 statement ok
-PREPARE hint_retry_stmt AS SELECT c FROM abc WHERE a = 10
+PREPARE hint_retry_stmt AS SELECT a FROM abc WHERE a = 10
 
 # First EXECUTE: abc_c_idx doesn't exist, so the hint should fail and fall
 # back to the unhinted plan.
@@ -2340,31 +2340,28 @@ isolation level: serializable
 priority: normal
 quality of service: regular
 ·
-• project
-│ columns: (c)
-│
-└── • scan
-      columns: (a, c)
-      sql nodes: <hidden>
-      kv nodes: <hidden>
-      regions: <hidden>
-      vectorized batch count: 0
-      KV time: 0µs
-      KV rows decoded: 0
-      KV pairs read: 0
-      KV bytes read: 0 B
-      KV gRPC calls: 0
-      estimated max memory allocated: 0 B
-      MVCC step count (ext/int): 0/0
-      MVCC seek count (ext/int): 0/0
-      actual row count: 0
-      estimated row count: 1 (missing stats)
-      table: abc@abc_pkey
-      spans: /10/0
+• scan
+  columns: (a)
+  sql nodes: <hidden>
+  kv nodes: <hidden>
+  regions: <hidden>
+  vectorized batch count: 0
+  KV time: 0µs
+  KV rows decoded: 0
+  KV pairs read: 0
+  KV bytes read: 0 B
+  KV gRPC calls: 0
+  estimated max memory allocated: 0 B
+  MVCC step count (ext/int): 0/0
+  MVCC seek count (ext/int): 0/0
+  actual row count: 0
+  estimated row count: 1 (missing stats)
+  table: abc@abc_pkey
+  spans: /10/0
 ·
 skipped statement hints: 1
  └── REWRITE INLINE HINTS
-     donor: SELECT c FROM abc@abc_c_idx WHERE a = _
+     donor: SELECT a FROM abc@abc_c_idx WHERE a = _
      skip reason: index "abc_c_idx" not found
 
 statement ok
@@ -2404,41 +2401,38 @@ isolation level: serializable
 priority: normal
 quality of service: regular
 ·
-• project
-│ columns: (c)
+• filter
+│ columns: (a)
+│ sql nodes: <hidden>
+│ regions: <hidden>
+│ vectorized batch count: 0
+│ execution time: 0µs
+│ actual row count: 0
+│ estimated row count: 1 (missing stats)
+│ filter: a = 10
 │
-└── • filter
-    │ columns: (a, c)
-    │ sql nodes: <hidden>
-    │ regions: <hidden>
-    │ vectorized batch count: 0
-    │ execution time: 0µs
-    │ actual row count: 0
-    │ estimated row count: 1 (missing stats)
-    │ filter: a = 10
-    │
-    └── • scan
-          columns: (a, c)
-          sql nodes: <hidden>
-          kv nodes: <hidden>
-          regions: <hidden>
-          vectorized batch count: 0
-          KV time: 0µs
-          KV rows decoded: 0
-          KV pairs read: 0
-          KV bytes read: 0 B
-          KV gRPC calls: 0
-          estimated max memory allocated: 0 B
-          MVCC step count (ext/int): 0/0
-          MVCC seek count (ext/int): 0/0
-          actual row count: 0
-          estimated row count: 1,000 (missing stats)
-          table: abc@abc_c_idx
-          spans: FULL SCAN
+└── • scan
+      columns: (a)
+      sql nodes: <hidden>
+      kv nodes: <hidden>
+      regions: <hidden>
+      vectorized batch count: 0
+      KV time: 0µs
+      KV rows decoded: 0
+      KV pairs read: 0
+      KV bytes read: 0 B
+      KV gRPC calls: 0
+      estimated max memory allocated: 0 B
+      MVCC step count (ext/int): 0/0
+      MVCC seek count (ext/int): 0/0
+      actual row count: 0
+      estimated row count: 1,000 (missing stats)
+      table: abc@abc_c_idx
+      spans: FULL SCAN
 ·
 applied statement hints: 1
  └── REWRITE INLINE HINTS
-     donor: SELECT c FROM abc@abc_c_idx WHERE a = _
+     donor: SELECT a FROM abc@abc_c_idx WHERE a = _
 
 statement ok
 SET tracing = off
@@ -2458,7 +2452,7 @@ statement ok
 DROP INDEX abc@abc_c_idx
 
 statement ok
-SELECT information_schema.crdb_delete_statement_hints('SELECT c FROM abc WHERE a = _')
+SELECT information_schema.crdb_delete_statement_hints('SELECT a FROM abc WHERE a = _')
 
 statement ok
 SELECT crdb_internal.await_statement_hints_cache()

